views/task-list.html

```diff
@@ -0,0 +1,331 @@
+                    <strong class="govuk-tag govuk-tag--grey app-task-list__tag" id="medical-information-status">Not checked</strong>
+                  {% endif %}
+                </li>
+                <li class="app-task-list__item">
+                  <span class="app-task-list__task-name">
+                    <a href="/confirmation-statement/statement-of-capital" aria-describedby="eligibility-status">
+                      Statement of capital
+                    </a>
+                  </span>
+                  {% if statementOfCapital === 'yes' %}
+                    <strong class="govuk-tag app-task-list__tag" id="eligibility-status">
+                      Checked</strong>
+                  {% elseif statementOfCapital === 'no' %}
+                    <strong class="govuk-tag govuk-tag--blue app-task-list__tag" id="eligibility-status">In progress</strong>
+                  {% else %}
+                    <strong class="govuk-tag govuk-tag--grey app-task-list__tag" id="medical-information-status">Not checked</strong>
+                  {% endif %}
+                </li>
+              </ul>
+            </li>
+            <li>
+              <h2 class="app-task-list__section">
+                People
+              </h2>
+              <ul class="app-task-list__items">
+                <li class="app-task-list__item">
+                  <span class="app-task-list__task-name">
+                    <a href="/confirmation-statement/active-officers" aria-describedby="eligibility-status">
+                      Officers
+                    </a>
+                  </span>
+                  {% if taskList.additionalOfficers === 'no' %}
+                    <strong class="govuk-tag app-task-list__tag" id="eligibility-status">
+                      Checked</strong>
+                  {% elseif taskList.additionalOfficers === 'yes' or officers === 'no' or activeOfficers === 'no' %}
+                    <strong class="govuk-tag govuk-tag--blue app-task-list__tag" id="eligibility-status">In progress</strong>
+                  {% else %}
+                    <strong class="govuk-tag govuk-tag--grey app-task-list__tag" id="medical-information-status">Not checked</strong>
+                  {% endif %}
+                </li>
+                <li class="app-task-list__item">
+                  <span class="app-task-list__task-name">
+                    {% if tradingStatus === '1' %}
+                      <a href="/confirmation-statement/active-pscs" aria-describedby="read-declaration-status">
+                        People with significant control (PSC)
+                      </a>
+                    </span>
+                    {% if taskList.pscStatement === 'yes' %}
+                      <strong class="govuk-tag app-task-list__tag" id="eligibility-status">
+                        Checked</strong>
+                    {% elseif taskList.additionalPscs === 'yes' or  taskList.pscs === 'no' or activePscs === 'no' %}
```

Comment:
the multiple tasklist checks should be removed, we should only pass a flag in to the template to toggle the status of a task.
basically we just want the html framework and styling of the page and the indicators that change (including the button disabled/enabled) should be passed in as flags/values from the controller. So the template will be told what to do by the controller. As we will be getting that info from teh api at a later date, it is ok to just pass in a dummy set of flags for this story.

Reply:
This might not be possible as we have:

 {% if taskList.sic === "yes" %}
            <strong class="govuk-tag app-task-list__tag" id="eligibility-status">
                Checked</strong>
        {% elseif taskList.sic === "no" %}
            <strong class="govuk-tag govuk-tag--blue app-task-list__tag" id="eligibility-status">In progress</strong>
        {% else %}
            <strong class="govuk-tag govuk-tag--grey app-task-list__tag" id="medical-information-status">Not checked</strong>
        {% endif %}

If we just have truthy or falsy then else and "Not Checked" will never be shown.

Reply:
an enumerated type would solve it

Reply:
just pass in the value of the enumerated type Checked/In progress/ Not checked

Reply:
I can go away and do something but it'd be good if we were on the same page can you give me an example of what form you think this should take, I'm presuming an enum with Checked,In progress, not checked

Reply:
Presumably we then replace:
completedTasks: string;
    email: string;
    exemption: string;
....

with

completedTasks: <enum_type>;
    email: <enum_type>;
    exemption:  <enum_type>;

Reply:
I was thinking of using an enum for TaskState. Each task has 3? states Checked/In progress/ Not checked. Then you'd have to use in the template {% if taskList.sicState === "Checked" %} then show 'checked' html

We might be able to define the html for Checked/In progress/ Not checked once each at top of template then insert the html for each depending on the state of the enum. So if we needed to show checked, we would use {{ set checkedHtml....  }.  to define the html then use {{checkedHtml}} to display it.  The html for the progress indicators seems to be the same
<strong class="govuk-tag govuk-tag--grey app-task-list__tag" id="medical-information-status">Not checked</strong>
but with different styling for each one.
Although not sure why they each have id="medical-information-status", you'd have to check with Oceanne if each should have a unique id and what they are used for.

the calculations (if a = yes and b = no etc) would get done before the page was rendered, so the template wouldn't have to work out if it should show checked/in progress/not checked, the template would just get told, "for sic task, show 'Checked'"

Reply:
I may not have understood this correctly, but I am unsure as to what we gain for the effort. I have just put up some refactored template code can we review this first?